data_validation/cli_tools.py

```
@@ -436,316 +395,351 @@ def _configure_validate_parser(subparsers):
+    optional_arguments.add_argument(
         "--threshold",
         "-th",
         type=threshold_float,
         help="Float max threshold for percent difference",
     )
-    custom_query_parser.add_argument(
+    optional_arguments.add_argument(
         "--use-random-row",
         "-rr",
         action="store_true",
         help="Finds a set of random rows of the first primary key supplied.",
     )
-    custom_query_parser.add_argument(
+    optional_arguments.add_argument(
         "--random-row-batch-size",
         "-rbs",
         help="Row batch size used for random row filters (default 10,000).",
     )
-    custom_query_parser.add_argument(
-        "--primary-keys",
-        "-pk",
-        help="Comma separated list of primary key columns 'col_a,col_b'",
-    )
-    custom_query_parser.add_argument(
+    optional_arguments.add_argument(
         "--wildcard-include-string-len",
         "-wis",
         action="store_true",
         help="Include string fields for wildcard aggregations.",
     )
-    custom_query_parser.add_argument(
+    optional_arguments.add_argument(
         "--cast-to-bigint",
         "-ctb",
         action="store_true",
         help="Cast any int32 fields to int64 for large aggregations.",
     )
 
+    # Group required arguments
+    required_arguments = custom_query_parser.add_argument_group("required arguments")
+    required_arguments.add_argument(
+        "--custom-query-type",
+        "-cqt",
+        required=True,
+        choices=["row", "column"],
+        help="Which type of custom query (row/column)",
+    )
+    required_arguments.add_argument(
+        "--source-query-file",
+        "-sqf",
+        required=True,
+        help="File containing the source sql query",
+    )
+    required_arguments.add_argument(
+        "--target-query-file",
+        "-tqf",
+        required=True,
+        help="File containing the target sql query",
+    )
+    required_arguments.add_argument(
```

PKs are only required for custom query row validations... maybe we should make it optional

---

Done.
Added a fail fast check in `cli_tools.py` by calling: `def apply_post_parse_checks(args: Namespace, parser) -> Namespace:`

---

PKs would be a required argument in the case of `validate custom-query row` command